edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Amqp/CbsNode.cs

```diff
@@ -73,24 +74,42 @@ public void RegisterLink(IAmqpLink link)
             Events.LinkRegistered(link);
         }
 
-        public async Task<bool> AuthenticateAsync(string id, Option<string> modelId)
+        public async Task<bool> AuthenticateAsync(string id, Option<string> modelId, Option<string> authChain)
         {
             try
             {
-                if (this.clientCredentialsMap.TryGetValue(id, out CredentialsInfo credentialsInfo))
+                // See if we received direct credentials for the target identity
+                CredentialsInfo credentialsInfo;
+                bool hasCredentials = this.clientCredentialsMap.TryGetValue(id, out credentialsInfo);
+
+                // Otherwise, this could be a child Edge acting OnBehalfOf of the target,
+                // in which case we would have the actor's credentials instead
+                if (!hasCredentials)
+                {
+                    Option<string> actorDeviceId = AuthChainHelpers.GetActorDeviceId(authChain);
+                    hasCredentials = actorDeviceId.Map(actor => this.clientCredentialsMap.TryGetValue(actor + $"/{CoreConstants.EdgeHubModuleId}", out credentialsInfo)).GetOrElse(false);
+                }
+
+                if (hasCredentials)
                 {
                     // Not -ve caching isAuthenticated here.
                     // If incorrect credentials are sent, then it authenticates every time and fails
                     // If correct credentials are sent later, then the authentication will succeed.
                     if (!credentialsInfo.IsAuthenticated)
                     {
-                        using (await credentialsInfo.AsyncLock.LockAsync())
+                        if (!credentialsInfo.IsAuthenticated)
```

Comment:
>if (!credentialsInfo.IsAuthenticated) [](start = 24, length = 37)

this is actually problematic, we're checking the authenticated status of the actor instead of the auth target.  So if the actor opens multiple links for different devices, only the first device will go through auth, and all others will end up skipping.